core/types/uuid.go

```diff
@@ -42,22 +43,48 @@ func (u UUID) String() string {
 	return uuid.UUID(u).String()
 }
 
-func (u UUID) Value() (driver.Value, error) {
-	return u[:], nil // []byte works for sql
+func (u *UUID) Value() (driver.Value, error) {
+	return u.String(), nil
```

Comment:
My understanding of this change is that this type no longer works with bytea, only a uuid or text in pg.  When used with a uuid column type, this won't actually be stored as a string in the DB storage, right?  This is just how we communicate it?
Even so, I'm a little uncertain why this has to be text instead of 16 bytes.

Reply:
For some reason when I was testing, I found that an error was fixed by making it string. Now that I am trying again, it seems like it works either way, so I will make it return `[]byte` again. I've tried so many different permutations of different things over the last 2 days I've forgotten why.

Reply:
I'm not sure why I said this. I am trying again, and confirmed using Bytes does not work. I get:

```
invalid input syntax for type uuid: "\xc430dbfc28a9508aaee9e6a2e1d81bb1" (SQLSTATE 22P02)
```

Reply:
Yeah, dang: https://github.com/jackc/pgx/blob/24c0a5e8ff740778bbed9810629cb83c9ea18bf9/pgtype/uuid.go#L91-L96
I guess that's just how `uuid_ossp` demands it.

Are we using any of the extension functions? 
 https://www.postgresql.org/docs/current/uuid-ossp.html#UUID-OSSP-FUNCTIONS-SECT
If not I'm wondering what this extension's `uuid` type on the backend gets us over bytea.  Strong typing in procedures i suppose doesn't work unless?

Reply:
>  Are we using any of the extension functions

Yeah, `uuid_generate_v5` is exposed to users. It's quite convenient, because it allows users to not have to guess on having available primary keys any more: https://github.com/kwilteam/kwil-db/blob/3a67ad32c8df6c8ba5d3c279992dfae43293d470/test/acceptance/test-data/users.kf#L88

Reply:
oh, yeah, nvm. This is hash based and txid is obv deterministic.

Reply:
Yeah, I'm just not quite sure how I could achieve that. This felt easier, and overall wasn't concerned with collisions.

Reply:
I'm just lost on why we need UUID everywhere.  Having a primary key column defined as `i SERIAL PRIMARY KEY` and letting it autoincrement is the easiest solution of all.
UUID caries risk, from performance to guarantees about the extension that provides it's functionality.
I don't much mind if schema authors really want it, but for internal tables it doesn't make a lot of sense to me.

Reply:
Auto-increment is not guaranteed to be deterministic. It's only guaranteed that it is greater than a previous value. Randal pointed this out to us, as he had seen people hit issues with the same assumption.

Reply:
I think this review comment can be resolved, but linking to a new discussion about the SERIAL issue should we decide to revisit: https://github.com/kwilteam/kwil-db/discussions/766